ionoscloud/resource_dbaas_mariadb_cluster.go

```diff
@@ -0,0 +1,245 @@
+				Required:    true,
+			},
+			"credentials": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Description: "Credentials for the database user to be created.",
+				Required:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"username": {
+							Type:             schema.TypeString,
+							Description:      "The username for the initial MariaDB user. Some system usernames are restricted (e.g 'mariadb', 'admin', 'standby').",
+							Required:         true,
+							ValidateDiagFunc: validation.ToDiagFunc(validation.All(validation.StringIsNotWhiteSpace, validation.StringLenBetween(1, 63))),
+						},
+						"password": {
+							Type:             schema.TypeString,
+							Description:      "The password for a MariaDB user.",
+							Required:         true,
+							Sensitive:        true,
+							ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
+						},
+					},
+				},
+			},
+			"maintenance_window": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Description: "A weekly 4 hour-long window, during which maintenance might occur.",
+				Optional:    true,
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						// TODO -- check if we need to add extra validation here for time
+						"time": {
+							Type:             schema.TypeString,
+							Description:      "Start of the maintenance window in UTC time.",
+							Required:         true,
+							ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
+						},
+						// TODO -- check if providing 'sunday' will lead to an error
+						"day_of_the_week": {
+							Type:             schema.TypeString,
+							Description:      "The name of the week day.",
+							Required:         true,
+							ValidateDiagFunc: validation.ToDiagFunc(validation.IsDayOfTheWeek(true)),
+						},
+					},
+				},
+			},
+			"dns_name": {
+				Type:        schema.TypeString,
+				Description: "The DNS name pointing to your cluster.",
+				Computed:    true,
+			},
+		},
+		Timeouts: &resourceDefaultTimeouts,
+	}
+}
+
+func mariaDBClusterCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(services.SdkBundle).MariaDBClient
+
+	cluster, err := dbaasService.GetMariaDBClusterDataCreate(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	response, _, err := client.CreateCluster(ctx, *cluster)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("an error occured while creating a DBaaS MariaDB cluster: %w", err))
+	}
+	clusterID := *response.Id
+	d.SetId(clusterID)
+
+	err = utils.WaitForResourceToBeReady(ctx, d, client.IsClusterReady)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error occured while checking the status for MariaDB cluster with ID: %v, error: %w", clusterID, err))
+	}
+	if err := client.SetMariaDBClusterData(d, response); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
```

Comment:
call `mariaDBClusterRead` on non-error return from CreateContext

Reply:
@cbeti-ionos is there a reason behind this? We are already setting the data into the state using `SetMariaDBClusterData`, not sure how an extra call to `mariaDBClusterRead` would help here.

Reply:
I think there might be a difference in the completeness of the data, since you use the response from a `client.CreateCluster` call and the cluster itself is not fully provisioned at this point.

Reply:
@cbeti-ionos `I think there might be a difference in the completeness of the data` hmm, it seems like a valid point, I will check if this is the case when the API will work, let's keep this open until then.

Reply:
I would also go a little bit further and say it would be a better separation of concerns to not have the Create function also set state data (wherever and whenever possible, we still need to set things like ID there), and delegate that responsibility entirely to the Reader and simply call it when we want state writing.

Reply:
@cbeti-ionos the API works now, there is no problem from the data completion POV, but let's discuss tomorrow about this.